src/logic/basic.lean

```diff
@@ -1424,62 +1408,67 @@ end nonempty
 lemma subsingleton_of_not_nonempty {α : Sort*} (h : ¬ nonempty α) : subsingleton α :=
 ⟨λ x, false.elim $ not_nonempty_iff_imp_false.mp h x⟩
 
+/-!
+### If-Then-Else
+
+lemmas about `ite` and `dite`
+-/
+
 section ite
+variables {α β γ : Sort*} {σ : α → Sort*} (f : α → β) (P Q : Prop) {p q : Prop} [decidable P]
+  [decidable Q] [decidable p] [decidable q] {a b c : α}
```

Comment:
Can you unify this section to use `P Q` instead of `p q` throughout?

Reply:
> change implicitness of some variables.

Then do this. I don't think we usually make a habit of using different variable names for implicit vs explicit variables.